src/server/detail/snapshot_storage.h

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+
+#pragma once
+
+#include <filesystem>
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include "io/io.h"
+#include "server/common.h"
+#include "util/cloud/aws.h"
+#include "util/fibers/fiberqueue_threadpool.h"
+#include "util/uring/uring_file.h"
+
+namespace dfly {
+namespace detail {
+
+namespace fs = std::filesystem;
+
+using namespace std::literals;
+
+constexpr std::string_view kS3Prefix = "s3://"sv;
```

Comment:
Now you do not need these in the header file.

Reply:
Its currently used by both `snapshotstorage.cc` and `save_stages_controller.cc` (will move more logic out of `save_stages_controller.cc` so should be able to remove from `save_stages_controller.cc` eventually)

Reply:
I think its fine, its really used in two placed. Additionally the only user of the storage header is the stages controller itself